### PR TITLE
Switch entities selection to a generic xml instance interface

### PR DESCRIPTION
**Pull request by shubham1g5**
https://github.com/dimagi/formplayer/pull/1125

### CODE REVIEW

**Comment by shubham1g5** [comment]
@snopoke bumping this for review as well. Thanks!

File: src/main/java/org/javarosa/core/reference/ReferenceHandler.java [hunk 1]
```diff
@@ -39,4 +39,9 @@ public static ReferenceManager instance() {
             return staticManager;
         }
     }
+
+    public static void clearInstance() {
+        threadLocalManager.set(null);
+        staticManager = null;
+    }
```
Comment (snopoke) [inline src/main/java/org/javarosa/core/reference/ReferenceHandler.java:46]: how is this change related?

**Review by snopoke — APPROVED** [review]
One question but otherwise looks good

File: src/main/java/org/javarosa/core/reference/ReferenceHandler.java (hunk 1 above)
Comment (shubham1g5) [reply src/main/java/org/javarosa/core/reference/ReferenceHandler.java:46]: This is to [fix tests in Formplayer](https://github.com/dimagi/formplayer/pull/1125/commits/e9d372a96180272184c9dd05984f52da5a512990).  For test cases using unarchived ccz file, we need to clear references to staticManager for app to correctly install.
